new_course/1_programming_fundementals/basic_data_types.md

```diff
@@ -0,0 +1,323 @@
+
+Strings can be changed, so every time a string is used, Ruby has to store it in memory even if an existing string with the same value already exists. Symbols, on the other hand, are stored in memory only once, making them faster in certain situations.
+
+One common application where symbols are preferred over strings are for the keys in hashes. We'll cover this in detail in the hashes lesson later in the course.
+
+You won't need to use symbols much in the beginning, but it's good to get familiar with what they are and what they look like so that you can recognize them.
+
+#### Create a Symbol
+To create a symbol, simply put a colon at the beginning of some text:
+```
+:my_symbol
+```
+
+#### Symbols vs. Strings
+To get a better idea of how symbols are stored in memory, give this a whirl in irb or a REPL. The [`#object_id` method](https://ruby-doc.org/core-2.6/Object.html#method-i-object_id) returns an integer identifier for an object. (And everything in Ruby is an object!)
+
+```ruby
+"string" == "string"  # => true
+
+"string".object_id == "string".object_id  # => false
+
+:symbol.object_id == :symbol.object_id    # => true
+```
+
+### Booleans
+You will learn about these data types in more detail in the conditional logic lesson later in this course. The goal in this lesson is get a basic understanding of what Booleans are.
+
+#### True and False
+The Boolean values `true` and `false` represent exactly what you think they do: `true` represents something that is true, and `false` represents something that is false.
+
+#### Nil
+'Nil' represents 'nothing'. Everything in Ruby has a return value. When a piece of code doesn't have anything to return, it will return `nil`. This is pretty abstract, but it will make more sense as you learn and use Ruby more.
+
+
+### Assignment
+
+1. Read the [Basics chapter](https://launchschool.com/books/ruby/read/basics) of LaunchSchool's *Introduction to Programming With Ruby* for a different explanation of Ruby's data types.
+2. Read through these Ruby Monstas sections about data types:
+    * [Numbers](http://ruby-for-beginners.rubymonstas.org/built_in_classes/numbers.html)
+    * [Strings](http://ruby-for-beginners.rubymonstas.org/built_in_classes/strings.html)
+    * [Symbols](http://ruby-for-beginners.rubymonstas.org/built_in_classes/symbols.html)
+    * [True, False, and Nil](http://ruby-for-beginners.rubymonstas.org/built_in_classes/true_false_nil.html)
+
+### Additional Resources
+This section contains helpful links to other content. It isn't required, so consider it supplemental for if you need to dive deeper into something.
+
+* If you want to go deeper into Ruby's numbers and string data types, read these chapters from the *Bastards Book of Ruby*:
+    * [Numbers](http://ruby.bastardsbook.com/chapters/numbers/)
+    * [Strings](http://ruby.bastardsbook.com/chapters/strings/)
+
+### Knowledge Check
```

Comment:
I _really_ like this new knowledge check section, great idea @leila-alderman. Thinking about how it will appear when on the site, do you think this should go above the additional resources section? that would also make it more consistent with the layout of older lessons.

Reply:
Glad you like it! I think it makes the beginning of the lessons nice and clean with the much shorter and punchier Learning Objectives. And I won't forget about going back up to quiz myself! 

My thinking in placing this section at the very end was that if you're reading through things and feel like you need more material to understand a concept, you're likely to want that before you quiz yourself. So the quiz is currently placed after all of the content, which also helps to more cleanly group all of the content/resources together. That said, I don't think it's a huge deal whether it comes before or after the Additional Resources. In terms of consistency with the layout of older lessons, though, I've been placing these sections after Additional Resources in the live lessons that I've been editing too (https://www.theodinproject.com/courses/web-development-101/lessons/git-basics). Let me know your thoughts!

Reply:
ah I didn't see that section on Git basics, it seems to fit well at the end of the lesson 😄

Reply:
Awesome, I'll keep placing those sections at the end then. Thanks for looking over these changes!